core/probe.cpp

```diff
@@ -294,7 +291,7 @@ Probe *GammaRay::Probe::instance()
 
 bool Probe::isInitialized()
 {
-    return instance() && qApp;
+    return instance();
```

Comment:
`instance()` now implies that a `qApp` exists, thus this check is no longer necessary. I updated the documentation of the Probe to make sure it's clear there's a strong coupling between Probe & QCoreApplication lifetime.